uaclient/entitlements/base.py

```diff
@@ -413,7 +447,7 @@ def can_enable(self) -> Tuple[bool, Optional[CanEnableFailure]]:
     # support additional reason types in the future.
     def enable(
         self,
-        silent: bool = False,
+        progress: api.ProgressWrapper,
```

Comment:
Is there a reason why we don't receive the `ProgressWrapper` object when creating the class instead of drilling down to the individual methods ?

Reply:
I have discussed that together and we agreed that the current solution is simpler